### ISSUE TITLE
etcd, api-server mesos tasks is unable to start on dcos v1.9.2

### ISSUE THREAD

**Issue by yleong** [mesosphere/dcos-kubernetes-quickstart#6]
tail of etcd stderr 
```
I0910 13:46:01.699224    36 exec.cpp:162] Version: 1.4.0
I0910 13:46:01.705013    29 exec.cpp:237] Executor registered on agent 6e686eb8-8059-422e-9123-9f709f6e0c1c-S54
2017-09-10 13:46:05.402293 I | pkg/flags: recognized and used environment variable ETCD_DATA_DIR=data-dir
2017-09-10 13:46:05.402559 I | pkg/flags: recognized and used environment variable ETCD_INITIAL_CLUSTER_TOKEN=kubernetes
2017-09-10 13:46:05.402692 I | pkg/flags: recognized and used environment variable ETCD_WAL_DIR=wal-pv/wal-dir
2017-09-10 13:46:05.402807 W | pkg/flags: unrecognized environment variable ETCD_LISTEN_CLIENT_PORT=2379
2017-09-10 13:46:05.402912 W | pkg/flags: unrecognized environment variable ETCD_LISTEN_PEER_PORT=2380
2017-09-10 13:46:05.403017 E | etcdmain: error verifying flags, expected IP in URL for binding (http://:2380). See 'etcd --help'.
```
tail of api-server stderr
```
I0910 13:46:26.444809    36 exec.cpp:162] Version: 1.4.0
I0910 13:46:26.450470    30 exec.cpp:237] Executor registered on agent 6e686eb8-8059-422e-9123-9f709f6e0c1c-S52
invalid argument "" for --bind-address=: failed to parse IP: ""
Usage of ./kube-apiserver:
```

I am trying to run the kubernetes framework on an existing DCOS v1.9.2. I used the `beta-kubernetes` package released on the docker image `mesosphere/universe-server:20170908T222335Z-version-3.x-3d0a4e4f33`

Does this release have any hard dependency on DCOS v1.10? 

marathon app.json for kubernetes framework
```
{
  "id": "/kubernetes",
  "connected": false,
  "recovered": false,
  "TASK_UNREACHABLE": 0,
  "cmd": "export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH; export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so); export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/); export JAVA_HOME=${JAVA_HOME%/}; export PATH=$(ls -d $JAVA_HOME/bin):$PATH &&  export JAVA_OPTS=\"-Xms256M -Xmx512M\" &&  ./kubernetes-scheduler/bin/kubernetes ./kubernetes-scheduler/svc.yml",
  "instances": 1,
  "cpus": 1,
  "mem": 1024,
  "disk": 0,
  "gpus": 0,
  "constraints": [],
  "fetch": [
    {
      "uri": "https://downloads.mesosphere.com/java/jre-8u144-linux-x64.tar.gz",
      "extract": true,
      "executable": false,
      "cache": false
    },
    {
      "uri": "https://downloads.mesosphere.com/kubernetes/assets/0.1.0-1.7.5-beta/kubernetes-scheduler.zip",
      "extract": true,
      "executable": false,
      "cache": false
    },
    {
      "uri": "https://downloads.mesosphere.io/libmesos-bundle/libmesos-bundle-1.10-1.4-63e0814.tar.gz",
      "extract": true,
      "executable": false,
      "cache": false
    }
  ],
  "storeUrls": [],
  "backoffSeconds": 1,
  "backoffFactor": 1.15,
  "maxLaunchDelaySeconds": 3600,
  "healthChecks": [
    {
      "gracePeriodSeconds": 900,
      "intervalSeconds": 30,
      "timeoutSeconds": 30,
      "maxConsecutiveFailures": 0,
      "portIndex": 0,
      "path": "/v1/plans/deploy",
      "protocol": "HTTP",
      "ignoreHttp1xx": false
    },
    {
      "gracePeriodSeconds": 900,
      "intervalSeconds": 30,
      "timeoutSeconds": 30,
      "maxConsecutiveFailures": 0,
      "portIndex": 0,
      "path": "/v1/plans/recovery",
      "protocol": "HTTP",
      "ignoreHttp1xx": false
    }
  ],
  "readinessChecks": [],
  "dependencies": [],
  "upgradeStrategy": {
    "minimumHealthCapacity": 0,
    "maximumOverCapacity": 0
  },
  "unreachableStrategy": {
    "inactiveAfterSeconds": 300,
    "expungeAfterSeconds": 600
  },
  "killSelection": "YOUNGEST_FIRST",
  "portDefinitions": [
    {
      "port": 0,
      "protocol": "tcp",
      "name": "api",
      "labels": {
        "VIP_0": "/api.kubernetes:80"
      }
    }
  ],
  "requirePorts": false,
  "labels": {
    "DCOS_COMMONS_UNINSTALL": "true",
    "DCOS_PACKAGE_RELEASE": "0",
    "DCOS_SERVICE_SCHEME": "http",
    "DCOS_PACKAGE_SOURCE": "http://kubernetes-universe.marathon.l4lb.thisdcos.directory/repo",
    "DCOS_PACKAGE_METADATA": "eyJwYWNrYWdpbmdWZXJzaW9uIjoiMy4wIiwibmFtZSI6ImJldGEta3ViZXJuZXRlcyIsInZlcnNpb24iOiIwLjEuMC0xLjcuNS1iZXRhIiwibWFpbnRhaW5lciI6InN1cHBvcnRAbWVzb3NwaGVyZS5jb20iLCJkZXNjcmlwdGlvbiI6IkhpZ2hseSBBdmFpbGFibGUgS3ViZXJuZXRlcyIsInRhZ3MiOlsia3ViZXJuZXRlcyJdLCJzZWxlY3RlZCI6ZmFsc2UsImZyYW1ld29yayI6dHJ1ZSwicHJlSW5zdGFsbE5vdGVzIjoiS3ViZXJuZXRlcyBvbiBEQy9PUyBpcyBjdXJyZW50bHkgaW4gQmV0YSBhbmQgc2hvdWxkIG5vdCBiZSB1c2VkIGluIHByb2R1Y3Rpb24uXG5cbkRlZmF1bHQgY29uZmlndXJhdGlvbiByZXF1aXJlcyAzIGFnZW50IG5vZGVzIGVhY2ggd2l0aDogNiBDUFUgfCA1NjU0IE1CIE1FTSB8IDYxNSBNQiBEaXNrXG5cblBsZWFzZSB2aXNpdCBvdXIgcXVpY2tzdGFydCByZXBvc2l0b3J5IChodHRwczovL2dpdGh1Yi5jb20vbWVzb3NwaGVyZS9kY29zLWt1YmVybmV0ZXMtcXVpY2tzdGFydCkgZm9yIGV4YW1wbGUgZGVwbG95bWVudHMuIiwicG9zdEluc3RhbGxOb3RlcyI6IkRDL09TIEt1YmVybmV0ZXMgaXMgYmVpbmcgaW5zdGFsbGVkIVxuXG5cdERvY3VtZW50YXRpb246IGh0dHBzOi8vZ2l0aHViLmNvbS9tZXNvc3BoZXJlL2Rjb3Mta3ViZXJuZXRlcy1xdWlja3N0YXJ0XG5cdElzc3VlczogaHR0cHM6Ly9qaXJhLm1lc29zcGhlcmUuY29tIiwicG9zdFVuaW5zdGFsbE5vdGVzIjoiREMvT1MgS3ViZXJuZXRlcyBoYXMgYmVlbiB1bmluc3RhbGxlZC4iLCJpbWFnZXMiOnsiaWNvbi1zbWFsbCI6Imh0dHBzOi8vZG93bmxvYWRzLm1lc29zcGhlcmUuY29tL2t1YmVybmV0ZXMvYXNzZXRzL2s4cy1zbWFsbC00OHg0OC5wbmciLCJpY29uLW1lZGl1bSI6Imh0dHBzOi8vZG93bmxvYWRzLm1lc29zcGhlcmUuY29tL2t1YmVybmV0ZXMvYXNzZXRzL2s4cy1tZWRpdW0tOTZ4OTYucG5nIiwiaWNvbi1sYXJnZSI6Imh0dHBzOi8vZG93bmxvYWRzLm1lc29zcGhlcmUuY29tL2t1YmVybmV0ZXMvYXNzZXRzL2s4cy1sYXJnZS0yNTZ4MjU2LnBuZyJ9fQ==",
    "DCOS_PACKAGE_REGISTRY_VERSION": "3.0",
    "DCOS_SERVICE_NAME": "kubernetes",
    "DCOS_PACKAGE_FRAMEWORK_NAME": "kubernetes",
    "DCOS_SERVICE_PORT_INDEX": "0",
    "DCOS_PACKAGE_VERSION": "0.1.0-1.7.5-beta",
    "DCOS_COMMONS_API_VERSION": "v1",
    "DCOS_PACKAGE_NAME": "beta-kubernetes",
    "MARATHON_SINGLE_INSTANCE_APP": "true",
    "DCOS_PACKAGE_IS_FRAMEWORK": "true"
  },
  "env": {
    "APISERVER_CPUS": "0.2",
    "KUBE_PROXY_MEM": "128",
    "ETCD_DISK_TYPE": "ROOT",
    "APISERVER_MEM": "512",
    "KUBERNETES_SERVICE_CIDR": "10.100.0.0/16",
    "KUBELET_MEM": "2048",
    "KUBERNETES_NODE_COUNT": "3",
    "KUBELET_CPUS": "2",
    "JAVA_URI": "https://downloads.mesosphere.com/java/jre-8u144-linux-x64.tar.gz",
    "CONTROLLERMANAGER_COUNT": "3",
    "KUBERNETES_VERSION": "v1.7.5",
    "LIBMESOS_URI": "https://downloads.mesosphere.io/libmesos-bundle/libmesos-bundle-1.10-1.4-63e0814.tar.gz",
    "ETCD_VERSION": "v3.2.6",
    "CONTROLLERMANAGER_CPUS": "0.2",
    "ETCD_PLACEMENT": "hostname:UNIQUE",
    "ETCD_CPUS": "0.2",
    "KUBERNETES_NODE_DISK": "1024",
    "SCHEDULER_COUNT": "3",
    "APISERVER_PLACEMENT": "hostname:UNIQUE",
    "ETCD_DISK": "512",
    "CONTROLLERMANAGER_PLACEMENT": "hostname:UNIQUE",
    "CONTROLLERMANAGER_MEM": "256",
    "APISERVER_COUNT": "3",
    "ETCD_COUNT": "3",
    "ETCD_MEM": "512",
    "SCHEDULER_CPUS": "0.2",
    "SCHEDULER_MEM": "256",
    "SCHEDULER_PLACEMENT": "hostname:UNIQUE",
    "EXECUTOR_URI": "https://downloads.mesosphere.com/dcos-commons/artifacts/0.30.0/executor.zip",
    "KUBE_PROXY_CPUS": "0.1",
    "FRAMEWORK_NAME": "kubernetes",
    "KUBERNETES_NODE_PLACEMENT": "hostname:UNIQUE",
    "FRAMEWORK_LOG_LEVEL": "INFO"
  }
}
```

**Comment by smugcloud** [comment]
@yleong this does [require](https://github.com/mesosphere/dcos-kubernetes-quickstart/blob/master/docs/existing_cluster.md) DC/OS 1.10.  Please upgrade your cluster, or create a 1.10 cluster and try it out.

**Comment by yleong** [comment]
Thanks for clarifying, you are right. The kubernetes scheduler uses `$MESOS_CONTAINER_IP` which is only available in Mesos 1.4/DCOS 1.10

But we still managed to modify the universe kubernetes to work on DCOS 1.9, in case anyone else is interested 
https://github.com/yleong/universe/tree/kubernetes-dcos1.9
